pkg/storage/base58wrapper/base58wrapper.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package base58wrapper
+
+import (
+	"encoding/base64"
+	"fmt"
+
+	"github.com/btcsuite/btcutil/base58"
+
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+)
+
+// package base58wrapper offers provider.Store wrappers for stores with certain constraints on key values such as
+// reserved prefix values (eg key IDs starting with underscore) by wrapping key values in another format supported by
+// the store.
+
+// NewBase58StoreWrapper creates a new Base64ToBase58StoreWrapper of store.
+func NewBase58StoreWrapper(store storage.Store) storage.Store {
+	return &Base64ToBase58StoreWrapper{store: store}
+}
+
+// Base64ToBase58StoreWrapper is a wrapper store that converts key IDs from base64 to base58 encoded values.
+// Keys IDs will be stored in the embedded store base58 encoded while the user of this wrapper store will interact with
+// the wrapper store using key IDs base64 raw (no padding) URL encoded.
+type Base64ToBase58StoreWrapper struct {
+	store storage.Store
+}
+
+func convert(k string) (string, error) {
+	kidBytes, err := base64.RawURLEncoding.DecodeString(k)
+	if err != nil {
+		return "", fmt.Errorf("convert: failed to base64 RAW URL Decode key: %w", err)
+	}
+
+	b58k := base58.Encode(kidBytes)
+
+	return b58k, nil
+}
+
+// Put stores the key and the record by converting the key from base64 to base58 encoded value first.
+func (b *Base64ToBase58StoreWrapper) Put(k string, v []byte) error {
+	b58k, err := convert(k)
+	if err != nil {
+		return fmt.Errorf("put: failed to convert key: %w", err)
+	}
+
+	return b.store.Put(b58k, v)
+}
+
+// Get fetches the record based on key by converting the key ID from base58 to base64 encoded value first.
+func (b *Base64ToBase58StoreWrapper) Get(k string) ([]byte, error) {
+	b58k, err := convert(k)
+	if err != nil {
+		return nil, fmt.Errorf("get: failed to convert key: %w", err)
+	}
+
+	return b.store.Get(b58k)
+}
+
+// Iterator returns an iterator for the latest snapshot of the underlying store.
+//
+// Args:
+//
+// startKey: Start of the key range, include in the range by converting it from base64 to base58 encoded value first.
+// endKey: End of the key range, not include in the range by converting it from base64 to base58 encoded value first.
+//
+// Returns:
+//
+// StoreIterator: a wrapped iterator for result range.
+func (b *Base64ToBase58StoreWrapper) Iterator(startKey, endKey string) storage.StoreIterator {
```

Comment:
If iterator is not working yet you should return not implemented error. Applies to all functions that are not working.

Reply:
I will return nil and add a comment here, it's too bad this interface function doesn't return error.

Reply:
done, and removed  the wrapperIterator.